### PR TITLE
feat: add search tweets

### PR DESCRIPTION
**Pull request by juliomerisio**
Adds search tweets functionality using MeiliSearch 

This PR uses the [Cron Job](https://console.cron-job.org/login) free service to visit the `/cron` route and populate the index every 5 minutes

I've hosted the MeiliSearch image on Digital Ocean, We have one year of free hosting there.

The intent is to add it as an Experimental Feature and see how it goes

 - ⚠️  it requires updating the environment variables

closes #262 

--- 
QA:  https://www.loom.com/share/5a48f3c736e841aca9e5ba06c3c05886

### CODE REVIEW

File: apps/web/src/components/Highlight.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+export type HighlightProps<THit> = {
+  hit: THit;
+  attribute: keyof THit | string[];
+  highlightedTagName?: React.ReactType;
+  nonHighlightedTagName?: React.ReactType;
+  className?: string;
+  separator?: string;
+};
+
+export function Highlight<THit extends AlgoliaHit<Record<string, unknown>>>({
```
Comment (juliomerisio) [inline apps/web/src/components/Highlight.tsx:47]: component used to create the highlighted word

File: apps/chrome-ext/src/popup/TweetComposer.tsx [hunk 2]
```diff
@@ -16,13 +16,15 @@ export const TweetComposer = () => {
...
         placeholder="Write your tweet concept/question here"
         value={text}
-        onChange={(e) => setText(e.target.value)}
+        onChange={e => setText(e.target.value)}
       />
 
       <Badge maxW="fit-content" colorScheme={counter < 0 ? 'red' : ''}>
         {counter}
       </Badge>
-      <Box as="span">cc @sseraphini</Box>
```
Comment (juliomerisio) [inline apps/chrome-ext/src/popup/TweetComposer.tsx:25]: note to myself - remove it from the ext

File: apps/web/README.md [hunk 3]
```diff
@@ -15,3 +15,8 @@ This is the Twitter API Bearer Token. You can get a new one with the following s
 - Create a new project
 - Create an app inside the project
 - After that you will have the API Key, the Bearer Token and the Access Token. In this case we will only need the Bearer Token. After this just add it to the .env file and you are all set.
+
+### MEILI_ACCESS_TOKEN
+
+This is the access token for the MeiliSearch API. You can get it following the quick guide:
+- [Quick start](https://docs.meilisearch.com/learn/getting_started/quick_start.html#step-1-setup-and-installation)
```
Comment (juliomerisio) [inline apps/web/README.md:22]: provide a step by step guide

File: apps/web/src/components/Timeline.tsx [hunk 4]
```diff
@@ -1,24 +1,59 @@
...
+import { AlgoliaHit } from 'instantsearch.js';
 
 interface Props {
-  initialTweets?: TweetData[];
+  initialTweets?: HitProps['hit'][];
   initialNextToken?: string;
   query?: string;
   isSearch?: boolean;
 }
+export type TweetData = AlgoliaHit<{
```
Comment (juliomerisio) [inline apps/web/src/components/Timeline.tsx:19]: export type from ./types folder

File: apps/web/src/components/Timeline.tsx [hunk 5]
```diff
@@ -78,11 +112,25 @@ export const Timeline = ({
     );
   };
 
+  const renderMeiliSearch = () => {
+    return (
+      <>
+        <style>
+          {`
```
Comment (juliomerisio) [inline apps/web/src/components/Timeline.tsx:119]: remove inline style tag

**Review by juliomerisio — COMMENTED** [review]
Need to update tests

File: apps/web/src/components/Highlight.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  hit,
+  attribute,
+  highlightedTagName = 'mark',
+  nonHighlightedTagName = 'span',
+  separator = ', ',
+  ...rest
+}: HighlightProps<THit>) {
+  const { value: attributeValue = '' } =
+    getPropertyByPath(hit._highlightResult, attribute as string) || {};
+  const parts = getHighlightedParts(attributeValue);
```
Comment (NicolasLopes7) [inline apps/web/src/components/Highlight.tsx:57]:
```suggestion
  const highlightedParts = getHighlightedParts(attributeValue);
```

Comment (juliomerisio) [reply apps/web/src/components/Highlight.tsx]: great call, updated

**Review by juliomerisio — COMMENTED** [review]
should provide a fallback in case os missing keys or unavailable meili search service

**Comment by juliomerisio** [comment]
I'll save this for a better time.
This feature shows to be a low priority right now.

**Comment by sibelius** [comment]
I've got some time to work and review these, can you reopen?

**Pull request closed without merging**